v3/src/components/case-table/row-drag-overlay.tsx

```diff
@@ -0,0 +1,54 @@
+import { DragOverlay, Modifier } from "@dnd-kit/core"
+
+// Define TRows type
+type TRows = {
+  __id__: string;
+  [key: string]: any;
+}
+import React from "react"
+import { kInputRowKey } from "./case-table-types"
+import DragIndicator from "../../assets/icons/drag-indicator.svg"
+
+import "./row-drag-overlay.scss"
+import styles from "./case-table-shared.scss"
+
+
+interface IProps {
+  rows: TRows[]
+  width: number | null
+}
+
+export function RowDragOverlay ({ rows, width }: IProps) {
+  const row = rows.find(r => r.__id__ === kInputRowKey)
+  const handleDropAnimation = () => {
+    /**
+     * If there has been no drop we would like to animate the overlay back to its original position.
+     * Otherwise, we don't want to animate it at all.
+     */
+  }
+
+  const restrictToVertical: Modifier = ({ transform }) => {
+    return {
+      ...transform,
+      x: 0, // Restrict horizontal movement
+    }
+  }
+
+  const style = { height: styles.bodyRowHeight, width: width ?? "100%" }
```

Comment:
This hardcodes the overlay height to the default. Do we want to show the current row height for the overlay?

Reply:
Fixed in last commit.